mjh_sanity/godas_obssanitycheck.py

```diff
@@ -0,0 +1,167 @@
+        ax.set_ylabel('Min/Max')
+        #ax.set_xlabel('Dates')
+        ax.legend(loc=0)
+        ax.xaxis.set_major_locator(mdates.DayLocator(interval=10))
+        #ax.xaxis.set_tick_params(rotation=40, labelsize=9)
+        ax.set_xticklabels([])
+        ax=fig.add_subplot(gs[:,1])
+        plt.title('missing files on dates:')
+        md_len=len(missing_dates)
+        print('number of missing files', md_len)
+        if md_len == 0:
+            plt.text(.12, .75, 'No missing files!', fontsize=13)
+            plt.text(.06, .65, 'from %s to %s'%(avail_dates[0], avail_dates[-1]))
+        else:
+            xstep=0.01
+            ystep=0.95
+            for tx in missing_dates:
+                plt.text(xstep, ystep, '%s'%tx, fontsize=8)
+                ystep=ystep-0.05
+                if ystep < .005: 
+                    xstep=xstep+0.33
+                    ystep=0.95
+        
+        ax.set_xticklabels([])
+        ax.set_yticklabels([])
+        plt.suptitle("File descriptor: %s"%file_descriptor, y=.95);
+        plt.savefig('Fig_%s_%s.png'%(file_descriptor, self.year), bbox_inches='tight', pad_inches=0.1)
+
+
+    def find_nlocs(self, filename):
+        args=self.args
+        nc_fid = Dataset(filename,'r')
+        nloc=nc_fid.variables['nlocs'].size
+        nc_fid.close()
+        return nloc
+
+    def find_minmax(self, fname):
+        args=self.args
+        ncfile = Dataset(fname,'r')
+        metadata_grp=ncfile.groups['MetaData']
+        var_grp=ncfile.groups[args.group]
+        var_tmp=np.squeeze(var_grp.variables[args.variable][:])
+        minval=min(var_tmp)
+        maxval=max(var_tmp)
+        ncfile.close()
+        return minval, maxval
+        
+if __name__ == '__main__':
+    description = """ Ex: godas_obssanitycheck.py -f adt_c2 -o missing -g ObsValue 
+                                       -v sea_surface_temperature -p /path/to/folder
+                  """
+    # Command line argument
+    parser = ArgumentParser(
+        description=description,
+        formatter_class=ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-f',
+        '--file_descriptor',
+        type=str, nargs='+',required=True)
+    parser.add_argument(
+        '-v',
+        '--variable',
+        help="ioda obs name (absolute_dynamic_topography, sea_surface_temperature, sea_surface_salinity, ...)",
+        type=str, required=False)
+    parser.add_argument(
+        '-p',
+        '--path',
+        help="path name",
+        type=str, required=False)
+    parser.add_argument(
+        '-s',
+        '--start',
+        help="start date in the format yearmmdd",
+        type=str, required=True)
+    parser.add_argument(
+        '-e',
+        '--end',
+        help="end date in the format yearmmdd",
+        type=str, required=True)
+    parser.add_argument(
+        '-g',
+        '--group',
+        help="ioda groups [ObsError, ombg, oman, ObsValue, ...]",
```

Comment:
we're only checking the obs, not the DA. Remove `ombg`, `oma`. But more importantly, why should the user need to know what group to use? The group should not be an argument, remove that option.